pkg/generators/golang/clients_generator.go

```diff
@@ -890,6 +890,13 @@ func (g *ClientsGenerator) generateRequestSource(method *concepts.Method) {
 			{{ end }}
 			return
 		}
+
+		// Impersonate wraps requests on behalf of another user.
+		// Note: Only clusters service API calls support impersonation.
```

Comment:
I have one concern with this approach:
All types of requests will now have an "Impersonate" method. However, only clusters-service checks, and supports the impersonation header. If someone were to call Impersonate() for another service that isn't checking for the impersonation header, whatever call that is being made will be run without error and without the user being impersonated. I added this Note: comment to try to mitigate these types of mistakes.